src/main/scala/com/fulcrumgenomics/fastq/SortFastq.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.fastq
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.sopt.{arg, clp}
+import com.fulcrumgenomics.util.{Io, Sorter}
+
+/** Companion object for the SortFastq tool. */
+object SortFastq {
+  private class FastqCodec extends Sorter.Codec[FastqRecord] {
+    /** Encodes the record directly to fastq. */
+    override def encode(rec: FastqRecord): Array[Byte] = rec.toString.getBytes
+
+    /** Decodes the record from the fastq string. */
+    override def decode(bs: Array[Byte], start: Int, length: Int): FastqRecord = {
+      val lines = new String(bs, start, length).lines
+      val name       = lines.next().substring(1)
```

Comment:
I know this is a little suspect in that it'll parse the entire header line into "name", including any read number and comment string, but since all we do is spit them back out another file it doesn't actually hurt.